### PR TITLE
improve error logging, fixes #1440

### LINKED ISSUE #1440: LockTimeout: better error message
When I have a locking problem, I would like to have a "normal" error message like:

Error: The repository is locked, please try again later, or use the option --lock-wait.

Instead of the horrible stack trace:
```
Remote: Borg 1.0.6: exception in RPC call:
Remote: Traceback (most recent call last):
Remote:   File "/home/xyz/borg-env/lib/python3.4/site-packages/borg/locking.py", line 136, in acquire
Remote:     os.mkdir(self.path)
Remote: FileExistsError: [Errno 17] File exists: '/home/xyz/repos/host/trunk/lock.exclusive'
Remote:
Remote: During handling of the above exception, another exception occurred:
Remote:
Remote: Traceback (most recent call last):
Remote:   File "/home/xyz/borg-env/lib/python3.4/site-packages/borg/remote.py", line 97, in serve
Remote:     res = f(*args)
Remote:   File "/home/xyz/borg-env/lib/python3.4/site-packages/borg/remote.py", line 128, in open
Remote:     self.repository.__enter__()  # clean exit handled by serve() method
Remote:   File "/home/xyz/borg-env/lib/python3.4/site-packages/borg/repository.py", line 82, in __enter__
Remote:     self.open(self.path, self.exclusive, lock_wait=self.lock_wait, lock=self.do_lock)
Remote:   File "/home/xyz/borg-env/lib/python3.4/site-packages/borg/repository.py", line 171, in open
Remote:     self.lock = UpgradableLock(os.path.join(path, 'lock'), exclusive, timeout=lock_wait).acquire()
Remote:   File "/home/xyz/borg-env/lib/python3.4/site-packages/borg/locking.py", line 265, in acquire
Remote:     with self._lock:
Remote:   File "/home/xyz/borg-env/lib/python3.4/site-packages/borg/locking.py", line 120, in __enter__
Remote:     return self.acquire()
Remote:   File "/home/xyz/borg-env/lib/python3.4/site-packages/borg/locking.py", line 141, in acquire
Remote:     raise LockTimeout(self.path)
Remote: borg.locking.LockTimeout: /home/xyz/repos/host/trunk/lock.exclusive
Remote: Platform: Linux xyz 3.13.0-76-generic #120-Ubuntu SMP Mon Jan 18 15:59:10 UTC 2016 x86_64 x86_64
Remote: Linux: Ubuntu 14.04 trusty
Remote: Borg: 1.0.6  Python: CPython 3.4.3
Remote: PID: 37645  CWD: /home/xyz/repos/host
Remote: sys.argv: ['/home/xyz/bin/borg', 'serve', '--restrict-to-path', '/home/xyz/repos/host']
Remote: SSH_ORIGINAL_COMMAND: 'borg serve --umask=077'
Remote:
Remote Exception (see remote log for the traceback)
Platform: Linux host 3.2.0-4-amd64 #1 SMP Debian 3.2.81-1 x86_64
Linux: debian 7.11
Borg: 1.0.6  Python: CPython 3.5.2
PID: 45908  CWD: /root/bin
sys.argv: ['borg', 'mount', 'xyz@backup.example.com:trunk::20160804', '/mnt']
SSH_ORIGINAL_COMMAND: None
```

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
archiver: split traceback and msg, have separate log level for traceback, log LockTimeout at debug level, for "Error" exceptions: always log the traceback, either at ERROR or DEBUG level.

remote: if we have an "Error" typed exception instance, we can use its traceback flag and .get_message() as we do locally.

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -2354,37 +2354,51 @@ def main():  # pragma: no cover
...
         exit_code = EXIT_ERROR
     except SIGTERMReceived:
-        msg = 'Received SIGTERM.'
+        msg = 'Received SIGTERM'
+        tb_log_level = logging.DEBUG
+        tb = '%s\n%s' % (traceback.format_exc(), sysinfo())
         exit_code = EXIT_ERROR
     if msg:
         logger.error(msg)
+    if traceback:
```
Comment (enkore) [inline src/borg/archiver.py]: `traceback` is a module. `e.traceback`?

File: src/borg/archiver.py [hunk 2]
```diff
@@ -2354,37 +2354,51 @@ def main():  # pragma: no cover
...
+        if tb_log_level == logging.ERROR:
+            print(tb, file=sys.stderr)
         sys.exit(e.exit_code)
     try:
         exit_code = archiver.run(args)
     except Error as e:
         msg = e.get_message()
-        if e.traceback:
-            msg += "\n%s\n%s" % (traceback.format_exc(), sysinfo())
+        tb_log_level = logging.ERROR if e.traceback else logging.DEBUG
```
Comment (enkore) [inline src/borg/archiver.py]: Referring to L2400; this is unnecessary. tb_log_level wouldn't be used if `e.traceback` isn't.

File: src/borg/archiver.py [hunk 3]
```diff
@@ -2354,37 +2354,51 @@ def main():  # pragma: no cover
...
-        msg = 'Keyboard interrupt.\n%s\n%s' % (traceback.format_exc(), sysinfo())
+        msg = 'Keyboard interrupt'
+        tb_log_level = logging.DEBUG
+        tb = '%s\n%s' % (traceback.format_exc(), sysinfo())
         exit_code = EXIT_ERROR
     except SIGTERMReceived:
-        msg = 'Received SIGTERM.'
+        msg = 'Received SIGTERM'
+        tb_log_level = logging.DEBUG
+        tb = '%s\n%s' % (traceback.format_exc(), sysinfo())
```
Comment (enkore) [inline src/borg/archiver.py]: Maybe deduplicate the identical `tb = '%s\n%s' % (traceback.format_exc(), sysinfo())` lines e.g. make this the default of `tb`

File: src/borg/archiver.py (hunk 2 above)
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: Before this changeset e.traceback meant "show a traceback" (log on ERROR level), if it was False, the traceback was not output at all. I changed that, it is emitted on DEBUG level now in that case, so one can see it if one runs borg on debug level.

File: src/borg/archiver.py (hunk 3 above)
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: Tried, but it doesn't get better by that. Maybe a bit less duplicate, but more complicated logic.

File: src/borg/archiver.py (hunk 2 above)
Comment (enkore) [inline src/borg/archiver.py]: Ah right. That confused me, I thought e.traceback would be the _actual_ traceback (but that's probably `e.__traceback__`). Then this looks correct.

**Comment by enkore** [comment]
lgtm

**Pull request merged**